extensions/darwin/clock-nanosleep.c

```diff
@@ -0,0 +1,80 @@
+/* Copyright (C) 2020-2022 Aleix Conchillo Flaqué <aconchillo@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA
+ */
+
+#include "clock-nanosleep.h"
+
+#include <errno.h>
+#include <time.h>
+#include <sys/types.h>
+
+/**
+ * The code below has been extracted from the following notcurses files:
+ *
+ * https://github.com/dankamongmen/notcurses
+ *
+ *   src/compat/compat.c
+ *   src/compat/compat.h
+ */
+
+#define NANOSECS_IN_SEC 1000000000ul
+
+#ifdef __APPLE__
+#define TIMER_ABSTIME 1
+#endif
+
+static inline uint64_t
```

Comment:
You are using uint64_t without importing `<stdint.h>`. Unless there is some spec guaranteeing that <sys/types.h> imples uint64_t, I propose to not do that.

Reply:
Added `<stdint.h>`